pkgs/applications/version-management/gittyup/default.nix

```diff
@@ -0,0 +1,96 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, cmark
+, darwin
+, git
+, libssh2
+, lua5_4
+, hunspell
+, ninja
+, openssl
+, pkg-config
+, qtbase
+, qttools
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gittyup";
+  version = "1.2.2";
+
+  src = fetchFromGitHub {
+    owner = "Murmele";
+    repo = "Gittyup";
+    rev = "gittyup_v${version}";
+    hash = "sha256-JJ20vls/NGkm0xV+vDguvuW5yqhOQf83TMvnn5Kx4IE=";
+    fetchSubmodules = true;
+  };
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=OFF"
+    "-DUSE_SYSTEM_CMARK=ON"
+    "-DUSE_SYSTEM_GIT=ON"
+    "-DUSE_SYSTEM_HUNSPELL=ON"
+    # "-DUSE_SYSTEM_LIBGIT2=ON" # upstream uses its own fork of libgit2
```

Comment:
1. Is this settable?
2. Why not use OFF instead of comment?

Reply:
The idea of leaving this flag commented is because maybe someday in the future it will make sense to use it (upstream is in the process of updating their fork of `libgit2`). But not the case right now.

Setting it to `OFF` without the comment doesn't make sense for the same reason.

Reply:
When I said "commented", I was referring to the whole line.

It makes few to no sense to comment a whole line of otherwise executable code - it looks like "I forgot to cleanup the debugging session".

Reply:
It makes if there is desire to enable it in the future. This is what I want to do it someday, since ideally we shouldn't be using embedded libraries from upstream.

Reply:
BTW, I think this comment is specially important because maybe someone in the future will try to enable this flag because they want to get rid from the embedded `libgit2`, only to suffer with the frustration of things getting broken.

So this commented code serves two purposes: to remember that in a future release this may be possible, but that right now it doesn't work.

Reply:
It still looks like "I forgot the cleanup before upload". 

Elaborate a better comment explaining why you are commenting an otherwise executable line.

Reply:
Forget. Do as you wish.

Reply:
Updated the comment at 82e3a36. Does this looks better?